### ISSUE TITLE
Client-side menus

### ISSUE THREAD

**Issue by PlusGit** [PlusGit/theta#41]
Yeah...

**Comment by PlusGit** [comment]
Yeah, seeing as there's not much else to do – and clientside menus are rather important – I'm gonna push this to the 1.00.0 milestone.

**Comment by PlusGit** [comment]
Only issue I've found so far is that sometimes the menu won't exit properly.

**Comment by PlusGit** [comment]
Well, it works pretty well so far!  
I tested in a local server and I found these issues:
 - (In multiplayer:) The menu thinks the player has 0 cash, meaning all items are greyed out except the free ones (but they're still purchasable). I could solve this by moving credits to an inventory item, which I've planned to do for a while.
 - If the above is fixed, the player's money wouldn't be updated until the next menu redraw. This could be fixed by calling `MSCC_Draw`/`MSCR_Draw` in [the credit management](https://github.com/PlusGit/theta/blob/master/src/code/acs_src/a_monies.acs) functions.
 - Some class item
s have a cyan label. Don't know why. This should only happen if the class has `CF_ADDON`:  
![screenshot_doom_20151204_075942](https://cloud.githubusercontent.com/assets/2996684/11584594/072ed020-9a5d-11e5-9930-0200e007fd36.png)  
This seems to occur randomly.

Also:
 - It doesn't look good on lower resolutions (I play on 1920*1080 usually so I designed it for that)

**Comment by Proluiz** [comment]
Revision patch b9fe86b3d2 .
Glitchs:
Buying shotgunner class give me lots of bugs and op stuffs.
![screenshot_doom_20160113_213022](https://cloud.githubusercontent.com/assets/6380147/12311242/0440319c-ba3d-11e5-8903-5ff5b92b3ed3.png)

Buying rifle also.
![screenshot_doom_20160113_213319](https://cloud.githubusercontent.com/assets/6380147/12311286/64d6d8f8-ba3d-11e5-9886-dc1513500f6b.png)

Buying addons also is spammable you sometimes cant buy (ever not having and sometimes you can buy 3 of them and they also random became avaliable)
![screenshot_doom_20160113_213637](https://cloud.githubusercontent.com/assets/6380147/12311401/f6924ae8-ba3d-11e5-9eda-101bcd0591ae.png)

also fixing something i told before everything you buy became you into admin, also something is working right, the resing option.

**Comment by PlusGit** [comment]
OK, I merged the `clientside-menus` branch into `master.` We need some final testing.
